File: functions_environment/detect_environment.py
import os
import socket

def detect_environment():
    #  Check hostname
    hostname = socket.gethostname()

    if 'Drakarykseni.local' in hostname:  # Replace with your local hostname
        return 'localNeus'
    elif 'local-hostname' in hostname:  # Replace with your server hostname
        return 'localAnttiL'
    else  # Server hostname can vary depending on the node, interactive seesion, etc.
        return 'server'

    raise EnvironmentError('Environment could not be detected.')

File: functions_environment/detect_environment.m
function env = detect_environment()
    % Detect the environment based on unique characteristics    
    % Check hostname
    [status, hostname] = system('hostname');
    hostname = strtrim(hostname); % Remove any trailing newline or spaces

    if status == 0
        if contains(hostname, 'local-hostname') || contains(hostname, 'xxxx')% Replace aaaa and xxxx with your local hostname
            env = 'localUserA';
            return;
        elseif contains(hostname, 'local-hostname') % Replace with your local hostname
                env = 'localUserB';
                return;
        else % Server hostname can vary depending on the node, etc.
            env = 'server';
            return;
        end
    end
end
